pyramid/tierFS.go

```diff
@@ -320,6 +304,10 @@ func (tfs *TierFS) newLocalFileRef(namespace, filename string) localFileRef {
 }
 
 func (tfs *TierFS) objPointer(namespace, filename string) block.ObjectPointer {
+	if runtime.GOOS == "windows" {
+		filename = strings.ReplaceAll(filename, `\\'`, "/")
```

Comment:
This change is fundamentally unsafe, e.g. if/when MacOS decides it has a new way of writing file paths.  I think that in all files that are not local, we should avoid `filepath.Join` in favour of URL-based joiners or our very own, and write our own `/`s.

Reply:
Not sure I follow - how can we prepare for a scenario when MacOS decides to use '^' as the separator? Filenames can contain the new separator.